client/my-sites/importer/importer-header.jsx

```diff
@@ -131,4 +146,6 @@ const mapDispatchToProps = dispatch => ( {
 	startImport: flowRight( dispatch, startImport ),
 } );
 
-export default connectDispatcher( null, mapDispatchToProps )( localize( ImporterHeader ) );
+export default connect( null, { recordTracksEvent } )(
```

Comment:
This seems off to me - we've essentially got 2 different `connect`s and  `mapDispatchToProps` here right?

```js
// 1st:
const mapDispatchToProps = dispatch => ( {
	startImport: flowRight( dispatch, startImport ),
} );


export default connect( 
  null, 
  // 2nd:
  { recordTracksEvent } 
)(
	connectDispatcher( 
      null, 
      // 2nd being used
      mapDispatchToProps
    )( localize( ImporterHeader ) )
);

Reply:
I don't completely understand the purpose of `connectDispatcher` - I'd have to take a better look there, but perhaps `{ recordTracksEvent } ` should be added to the original `mDtP`?

If not I'd love to see this composed with `flow` to avoid the `c( b( a() ) )` wrapping pattern.

Reply:
This is an annoyance that comes out of the Flux storage that's powering the import section.

It's built to follow redux-like style, but it's still a separate dispatcher. In order to have both available, I had to wrap them in both dispatchers 😱 

Keeping it this way would allow easier refactoring when the Flux store gets thrown out and replaced with proper Redux code.

Reply:
confirmed: this was to allow incremental changeover from Flux to Redux. it's a bit awkward but it works. sorry it was never finished for you.

Reply:
That's fair, thanks for the explanation both! :)